### ISSUE TITLE
Using fetch and Safari

### ISSUE THREAD

**Issue by jetmirshatri** [zeit/next.js#756]
While developing an app I've used a custom server setup with `express` and I also used `fetch`. 
I had no idea that the latest version of Safari doesn't yet support `fetch` natively and it took me a while to understand this, so maybe it's possible to add a more friendlier error message?
I ended up using `isomorphic-fetch` 💯 

Error message running in dev mode (`const app = next({ dir: '.', dev: true });`) : 

<img width="984" alt="screen shot 2017-01-12 at 10 16 03" src="https://cloud.githubusercontent.com/assets/23201486/21912281/3c7220f4-d925-11e6-8f16-93fd14be41ad.png">

Error message running in prod mode (`const app = next({ dir: '.', dev: false });`) : 

<img width="720" alt="screen shot 2017-01-12 at 01 21 38" src="https://cloud.githubusercontent.com/assets/23201486/21912282/3dbd5fc8-d925-11e6-92d9-d2a4076a6126.png">

**Comment by arunoda** [comment]
In the production, we this is the accepted behaviour. (If something happens in the server).
If you provide us a sample app, we could have a look at it.

**Comment by timneutkens** [comment]
@arunoda managed to reproduce this, it's a larger issue than only fetch. Any exception when thrown in getInitialProps will give that callstack.

`pages/index.js`:

```js
import React from 'react'
import Link from 'next/link'

export default class extends React.Component {
  static async getInitialProps({ req }) {
    if(!req) {
      throw new Error('Error message')
    }
    return {}
  }

  render() {
    return <div><Link href="/test">example</Link></div>
  }
}
```

`pages/test.js`:

```js
import React from 'react'
import Link from 'next/link'

export default class extends React.Component {
  static async getInitialProps({ req }) {
    if(!req) {
      throw new Error('Error message')
    }
    return {}
  }

  render() {
    return <div><Link href="/">example</Link></div>
  }
}
```

**Comment by arunoda** [comment]
@timneutkens So it's working fine in Chrome. Only if Safari?

**Comment by timneutkens** [comment]
@arunoda seems safari gives worse stacktraces, chrome gives the same one when throwing an error with the error message. I'll have a look into it so you can focus on more pressing issues 👍

**Comment by posrix** [comment]
Does this bug fixed yet? I still have this issue on next 4.2.3
